### DISCUSSION TITLE
Building service with user's auth required

### DISCUSSION THREAD

**Discussion by JamesWong1999** [supabase/supabase#3645] [Questions]
This may be a silly question. If I am using the service token (admin) on the service and each API should be only used for authentication users. The question is: do I need to invoke `const user = await getUser(token);` on each request. 

Relavant example: 
https://github.com/vercel/nextjs-subscription-payments/blob/main/pages/api/create-portal-link.js#L11

**Comment by goldcoders** [comment]
i think you need to save the token in local storage , then check if it is present if not invoke that function

**Reply by JamesWong1999** [reply]
thanks, @goldcoders isn't the service still to validate that token?
